File: Lib/Telerik/Examples/QuickStart/PropertyGrid/FirstLook/SurfaceElement.vb
Imports System.Text
Imports Telerik.WinControls
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.PropertyGrid.FirstLook
	Public Class SurfaceElement
		Inherits LightVisualElement
'INSTANT VB NOTE: The variable currentObject was renamed since Visual Basic does not allow class members with the same name:
		Private currentObject_Renamed As SurfaceObject

		Public Event CurrentObjectChanged As EventHandler

		Public Property CurrentObject() As SurfaceObject
			Get
				Return currentObject_Renamed
			End Get
			Set(ByVal value As SurfaceObject)
				If currentObject_Renamed IsNot value Then
					currentObject_Renamed = value
					Me.Children.Remove(currentObject_Renamed)
					Me.Children.Add(currentObject_Renamed)
					RaiseEvent CurrentObjectChanged(Me, EventArgs.Empty)
				End If
			End Set
		End Property

		Protected Overrides Function ArrangeOverride(ByVal finalSize As SizeF) As SizeF
			For Each element As SurfaceObject In Me.Children
				element.Arrange(New RectangleF(element.Offset.X, element.Offset.Y, element.DesiredSize.Width, element.DesiredSize.Height))
			Next element
			Return finalSize
		End Function
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/PropertyGrid/FirstLook/SurfaceElement.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls;
using Telerik.WinControls.UI;
using System.Drawing;

namespace Telerik.Examples.WinControls.PropertyGrid.FirstLook
{
    public class SurfaceElement : LightVisualElement
    {
        SurfaceObject currentObject;

        public event EventHandler CurrentObjectChanged;

        public SurfaceObject CurrentObject
        {
            get
            {
                return currentObject;
            }
            set
            {
                if (currentObject != value)
                {
                    currentObject = value;
                    this.Children.Remove(currentObject);
                    this.Children.Add(currentObject);
                    if (CurrentObjectChanged != null)
                    {
                        CurrentObjectChanged(this, EventArgs.Empty);
                    }
                }
            }
        }

        protected override SizeF ArrangeOverride(SizeF finalSize)
        {
            foreach (SurfaceObject element in this.Children)
            {
                element.Arrange(new RectangleF(element.Offset.X, element.Offset.Y, element.DesiredSize.Width, element.DesiredSize.Height));
            }
            return finalSize;
        }
    }
}
